### ISSUE TITLE
Use pyunit's skip functionality when available (i.e. Python > 2.6)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6006]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6006|
|Type|enhancement|
|Created|2012-09-18 17:02:21Z|
|Branch|https://github.com/twisted/twisted/tree/skip-pyunit-6006|

On Python 2.7 and 3.3, Python's unittest supports skipping. When trial has a test marked as skipped, it should use that support if possible so that non-trial runners can still deal with skips.

Python 3.3 needs this since otherwise skips are treated as failures.

<details><summary>Searchable metadata</summary>

```
trac-id__6006 6006
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_skip_pyunit_6006 branches/skip-pyunit-6006
branch_author__itamar itamar
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1347987741000000 1347987741000000
changetime__1348161632000000 1348161632000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35744]) Branching to 'skip-pyunit-6006'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review:
http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/skip-pyunit-6006

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @itamarst|
|-|-|

[1]
```
+if hasattr(pyunit, "SkipTest"):
```

I've been trained to fear hasattr and prefer getattr(... None) is not None, but I guess it's not a big deal with modules.

Clean and simple, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [35751]) Merge skip-pyunit-6006: pyunit test runner support trial skips.

Author: itamar
Review: therve
Fixes: [#6006](https://github.com/twisted/twisted/issues/6006)

Newer versions of Python's unittest have skip support, which can now recognize trial test case skips.
